ec2/spark_ec2.py

```diff
@@ -423,13 +426,14 @@ def get_spark_ami(opts):
         b=opts.spark_ec2_git_branch)
 
     ami_path = "%s/%s/%s" % (ami_prefix, opts.region, instance_type)
+    reader = codecs.getreader("ascii")
     try:
-        ami = urlopen(ami_path).read().strip()
-        print("Spark AMI: " + ami)
+        ami = reader(urlopen(ami_path)).read().strip()
```

Comment:
Yep.  The str->bytes gotcha remains the major stumbling block of py2->3 portability.  It is also worth note that more people use 3.* than 2.6.  

If I were you all, I would make a strong case for dropping 2.6 as 3 support comes online.